pkg/api/handler.go

```diff
@@ -157,6 +158,14 @@ func extractType(element interface{}) string {
 	v := reflect.ValueOf(element).Elem()
 	for i := 0; i < v.NumField(); i++ {
 		field := v.Field(i)
+
+		if field.Kind() == reflect.Map && field.Type().Elem() == reflect.TypeOf(dynamic.PluginConf{}) {
+			keys := field.MapKeys()
+			if len(keys) == 1 {
```

Comment:
can `keys` only have 0 or 1 entries? is it possible to have 2?

what about:
```
if keys := field.MapKeys(); len(keys) > 0 {
  return keys[0].String()
}
```

Reply:
If misconfigured it can indeed be 0 or greater than 1. But if there's more than one plugin in the `PluginConf` it's going to be considered invalid. So, I think we shouldn't return any type for this middleware.

Reply:
```suggestion
if keys := field.MapKeys(); len(keys) == 1 {
  return keys[0].String()
}```